daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/LossyValueTranslator.scala

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package engine
+package preprocessing
+
+import com.daml.lf.data.Ref.IdString
+import com.daml.lf.data._
+import com.daml.lf.language.Ast._
+import com.daml.lf.speedy.{ArrayList, SValue}
+import com.daml.lf.value.Value
+import com.daml.lf.value.Value._
+
+/** This translator is only intended for use where a SValue is needed for error reporting but
```

Comment:
can you link the issue I ask you to create lower.

Reply:
Could we move this to Daml script to be sure nobody else use it.

Reply:
Following a conversation with @remyhaemmerle-da it was decided to leave it closer to `ValueTranslator`.

Reply:
Updated comment to reference the daml issue.